apis/python/tests/test_indexed_dataframe.py

```diff
@@ -105,17 +101,49 @@ def simple_indexed_data_frame(tmp_path):
     }
     n_data = len(data["index"])
     rb = pa.Table.from_pydict(data)
-    sdf.write(rb)
-    yield (schema, sdf, n_data, index_column_names)
-    sdf.delete()
+    sidf.write(rb)
+    yield (schema, sidf, n_data, index_column_names)
+    sidf.delete()
+
+
+@pytest.fixture
+def doubly_indexed_data_frame(tmp_path):
```

Comment:
highly recommend testing more than two dimensions.  See the NdArray test fixtures for an example that does 1 to 5 dimensions as a parameter (really any number, but that seemed suffiicient).